### PR TITLE
[project-base] improved product edit in grapesjs

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Product editing in grapesjs now uses tag-input field to enter catnums. The design of the preview of products in the editor was improved. Also, grapesjs parser now uses DomCrawler instead of regex parsing.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://mg-textarea-grapesjs.odin.shopsys.cloud
  - https://cz.mg-textarea-grapesjs.odin.shopsys.cloud

### CODE REVIEW

**Review by vitek-rostislav — APPROVED** [review]

**Review by TomasLudvik — APPROVED** [review]

File: project-base/app/assets/js/admin/grapesjs/plugins/grapesjs-products-plugin.js [hunk 1]
```diff
@@ -74,9 +74,17 @@ export default grapesjs.plugins.add('products', editor => {
...
+            const templateName = productName === unkonwnProductName ? 'grapejs-unknown-product-card' : 'grapejs-product-card';
+
             element.appendChild(createDiv(`
-                <div style="margin-top: 2em; text-align: center;">${productName}<br/><br/>${product}</div>
-                <img src="${window.location.origin}/public/admin/images/grapejs-product-list-item.png" alt="${dataProductName}" style="max-width: 100%;/>
+                <div style="position: relative; max-width: 100%;">
+                    <div class="gjs-products-product-card-name">
+                        ${productName.substring(0, 55)}
+                        <br><br>
+                        ${product}
```
Comment (vitek-rostislav) [inline project-base/app/assets/js/admin/grapesjs/plugins/grapesjs-products-plugin.js]: nitpick - I would rename the variable to `productCatnum` to make it more clear

File: packages/framework/src/Component/GrapesJs/GrapesJsParser.php [hunk 2]
```diff
@@ -24,19 +24,28 @@ public function parse(?string $text): ?string
             return null;
         }
 
-        $newText = preg_replace_callback(static::GJS_PRODUCTS_REGEX, static function (array $matches): string {
-            $gjsProductsDiv = str_replace(["\r", "\n", "\r\n"], '', $matches[0]);
-            preg_match('/data-products="(.+?)"/', $gjsProductsDiv, $productMatches);
-            $productArray = explode(',', $productMatches[1]);
-            $trimmedProductArray = array_map(static fn ($product) => trim($product), $productArray);
+        $crawler = new Crawler($text);
```
Comment (vitek-rostislav) [inline packages/framework/src/Component/GrapesJs/GrapesJsParser.php:27]: This is awesome! :green_heart: 

Would it be too daring of me if I suggested there should be tests for `GrapesJsParser`? :smile:

**Review by vitek-rostislav — APPROVED** [review]

File: project-base/app/assets/js/admin/grapesjs/plugins/grapesjs-products-plugin.js (hunk 1 above)
Comment (grossmannmartin) [reply project-base/app/assets/js/admin/grapesjs/plugins/grapesjs-products-plugin.js]: renamed, good idea

File: packages/framework/src/Component/GrapesJs/GrapesJsParser.php (hunk 2 above)
Comment (grossmannmartin) [reply packages/framework/src/Component/GrapesJs/GrapesJsParser.php:27]: Tests rules! I fixed one error in implementation thanks to the test 💚

Comment (vitek-rostislav) [reply packages/framework/src/Component/GrapesJs/GrapesJsParser.php:27]: awesome! :rocket:

**Review by vitek-rostislav — APPROVED** [review]

**Review by pk16011990 — APPROVED** [review]

**Pull request merged**